### PR TITLE
Fix deposit block slot before genesis state

### PR DESCRIPTION
**Pull request by princesinha19**
This PR fixes the incorrect result for depositBlockSlot() before the genesis state.

[Resolves] #4465

### CODE REVIEW

**Comment by mcdee** [comment]
There are some additional nuances that need to be considered here:

  - there will be a number of validators processed at genesis, but not all of them.  The actual number will be `params.BeaconConfig().MinGenesisActiveValidatorCount` *plus* any additional validators in the Ethereum 1block this number was achieved (note that it is possible for there to be multiple deposits for the same validator)
  - post-genesis there can be a *maximum* of `params.BeaconConfig().MaxDeposits` deposits per slot
    - as such, the deposit inclusion slot can increase if a block is either not proposed or proposed without the maximum number of deposits

That said, the function should remain relatively simple because it can be called many times.  If will result in too high a CPU load it may make sense for the pre-genesis version the calculation to ignore the additional validators in the Ethereum 1 block after `MinGenesisActiveValidatorCount` (although in this case a note should be provided explaining why).

File: beacon-chain/rpc/validator/status.go [hunk 1]
```diff
@@ -188,8 +189,13 @@ func (vs *Server) depositBlockSlot(ctx context.Context, eth1BlockNumBigInt *big.
 	timeToInclusion := eth1UnixTime.Add(votingPeriodSeconds)
 
 	eth2Genesis := time.Unix(int64(beaconState.GenesisTime), 0)
-	eth2TimeDifference := timeToInclusion.Sub(eth2Genesis).Seconds()
-	depositBlockSlot := uint64(eth2TimeDifference) / params.BeaconConfig().SecondsPerSlot
+
+	if eth2Genesis.After(timeToInclusion) {
```
Comment (terencechain) [inline beacon-chain/rpc/validator/status.go]: This is a good candidate for a regression test

Comment (princesinha19) [reply beacon-chain/rpc/validator/status.go]: Yes, correct @terencechain. I have added the test cases for depostBlockSlot function. Please take a look.

**Comment by princesinha19** [comment]
@terencechain Can you please review this?

**Comment by terencechain** [comment]
@princesinha19 thanks for adding the test, it looks good.

e2e tests failed, something to look into

**Comment by princesinha19** [comment]
@terencechain It's working now. Actually it was not an issue. I don't know exactly why the e2e test failed with error:-

`eth1.go:67: mining log not found, this means the eth1 chain had issues starting: could not find requested text "Commit new mining work" in logs:
`

**Comment by princesinha19** [comment]
@terencechain Should we merge this?

**Pull request merged**